tools/loader/ImageClassifier.cpp

```diff
@@ -469,18 +565,43 @@ int main(int argc, char **argv) {
       inputImageData.convertToType(ElemKind::Float16Ty);
     }
 
+    // If we are benchmarking we are done with the while loop.
+    if (iterationsOpt) {
+      break;
+    }
     // About to run inference, so update the input image Placeholder's backing
     // Tensor with inputImageData.
     updateInputPlaceholders(bindings, {inputImagePH}, {&inputImageData});
 
     // Perform the inference execution, updating SMT.
     auto batchSize = inputImageData.dims()[0];
     loader.runInference(bindings, batchSize);
-
     // Print the top-k results from the output Softmax tensor.
     numErrors += processAndPrintResults(SMT, inputImageBatchFilenames);
   }
 
+  if (iterationsOpt) {
+    // Image tensors loaded up to be run at once for benchmark mode.
+    std::vector<std::unique_ptr<Tensor>> preLoadedImages;
```

Comment:
I think I might be missing something, but it seems like this isn't actually used anywhere? It's passed into `setupContextPool()` which adds tensors to it, but otherwise isn't used anywhere right? Once you call `updateInputPlaceholders()` the Placeholder has its own backing Tensor assigned from that Tensor. In fact inside `setupContextPool()` you could probably just insert `newImage` into the `PlaceholderBindings`, right? Either way not sure this is necessary.